File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/MailAlertRequest.scala
package io.swagger.server.model


/**
 * Payload for creating a mail alert.
 *
 * @param `type` Alert type identifier.
 * @param value Alert value or threshold.
 * @param to Email address to notify.
 * @param enabled Whether the alert is enabled.
 */
case class MailAlertRequest (
  `type`: Option[String],
  value: Option[String],
  to: Option[String],
  enabled: Option[String]
)

File: OPENAPI-REST-API/openapi-client/php-dt/src/App/DTO/MailAlertRequest.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

/**
 * Payload for creating a mail alert.
 */
class MailAlertRequest
{
    /**
     * Alert type identifier.
     */
    #[DTA\Data(field: "type", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $type = null;

    /**
     * Alert value or threshold.
     */
    #[DTA\Data(field: "value", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $value = null;

    /**
     * Email address to notify.
     */
    #[DTA\Data(field: "to", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $to = null;

    /**
     * Whether the alert is enabled.
     */
    #[DTA\Data(field: "enabled", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $enabled = null;

}
